### PR TITLE
Add get_copy argument in  get()  and related methods.

### PR DESCRIPTION
**Pull request by Nilanshrajput**
## Description
Add get_copy argument in ` get ` and related methods for getting objects without removing the remote object.
This was a todo listed in get() method (https://github.com/OpenMined/PySyft/blob/1ead29ce04d6ced3e08c3bd2c48c66851ffa50cf/syft/generic/pointers/object_pointer.py#L249)

This will enable persisting remote tensors and PySyft objects.

This is required in SyferText to host permanent Pipelines and SyferText objects on Pygrid Nodes

## Affected Dependencies
None

## How has this been tested?
No tests added

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: syft/workers/message_handler.py [hunk 1]
```diff
@@ -196,15 +196,18 @@ def respond_to_obj_req(self, msg: ObjectRequestMessage):
...
+        get_copy = msg.get_copy
 
         obj = self.get_obj(obj_id)
 
         permitted = all(map_chain_call(obj, "allow", user=user))
         if not permitted:
             raise GetNotPermittedError()
-        else:
+        elif not get_copy:
             self.object_store.de_register_obj(obj)
```
Comment (gmuraru) [inline syft/workers/message_handler.py:207]: MAYBE: Leave only one return obj at the end.

File: syft/generic/pointers/pointer_tensor.py [hunk 2]
```diff
@@ -385,12 +384,15 @@ def get(self, user=None, reason: str = "", deregister_ptr: bool = True):
...
                 this method is to move the tensor from the remote machine to the
                 local one, at which time the pointer has no use.
+            get_copy (bool): Setting get_copy True doesn't destroy remote.
 
         Returns:
             An AbstractTensor object which is the tensor (or chain) that this
             object used to point to #on a remote machine.
         """
-        tensor = ObjectPointer.get(self, user=user, reason=reason, deregister_ptr=deregister_ptr)
+        tensor = ObjectPointer.get(
```
Comment (gmuraru) [inline syft/generic/pointers/pointer_tensor.py:393]: BUG: You need to change the serde serialization tests to take into consideration the new parameter.
Also, there is an error here
```
return ObjectRequestMessage(obj_id=obj_id, user=None, reason=proto_msg.reason)
E       TypeError: __init__() missing 1 required positional argument: 'get_copy'
```

**Review by gmuraru — APPROVED** [review]

**Review by gmuraru — CHANGES_REQUESTED** [review]
Small things, but it looks good. I think a next step would be to replace everywhere in the code the ```copy().get()``` methods to only a ```get()``` with this param

**Comment by Nilanshrajput** [comment]
I solved one of the test failings but for this `test_serde_roundtrip_protobuf` i will have to update syft-proto 
```
@staticmethod
    def bufferize(worker, msg):
        """
        This method serializes a ObjectRequestMessage using ObjectRequestMessagePB.

        Args:
            msg (ObjectRequestMessage): input ObjectRequestMessage to be serialized.

        Returns:
            proto_msg (ObjectRequestMessagePB): serialized ObjectRequestMessage.
        """
        proto_msg = ObjectRequestMessagePB()
        sy.serde.protobuf.proto.set_protobuf_id(proto_msg.object_id, msg.object_id)
        proto_msg.reason = msg.reason
        return proto_msg

````
And ObjectRequestMessagePB() this is very complex I cant understand how its working, 
One solution is to pass default False here 
```
@staticmethod
    def unbufferize(worker, proto_msg):
        """
        This method deserializes ObjectRequestMessagePB into ObjectRequestMessage.

        Args:
            protobuf_msg (ObjectRequestMessagePB): input serialized ObjectRequestMessagePB.

        Returns:
           ObjectRequestMessage: deserialized ObjectRequestMessagePB.
        """
        obj_id = sy.serde.protobuf.proto.get_protobuf_id(proto_msg.object_id)
        # add worker support when it will be available
        return ObjectRequestMessage(obj_id=obj_id, user=None, reason=proto_msg.reason, get_copy=False)

```

File: syft/workers/message_handler.py (hunk 1 above)
Comment (Nilanshrajput) [reply syft/workers/message_handler.py:207]: Done

File: syft/generic/pointers/pointer_tensor.py (hunk 2 above)
Comment (Nilanshrajput) [reply syft/generic/pointers/pointer_tensor.py:393]: Done

**Comment by Nilanshrajput** [comment]
@gmuraru  I will remove copy in other PR (I will create an issue for that now) this one's little urgent for syfertext

**Comment by gmuraru** [comment]
> @gmuraru I will remove `copy().get()` in other PR (I will create an issue for that now) this one's little urgent for syfertext

Yep - there was also a stale PR that was opened before - but I think there was no review for it :\

File: syft/workers/message_handler.py [hunk 3]
```diff
@@ -196,16 +196,19 @@ def respond_to_obj_req(self, msg: ObjectRequestMessage):
...
             return obj
```
Comment (gmuraru) [inline syft/workers/message_handler.py:208]: NITPICK: Remove this since you return the object at the end

Comment (Nilanshrajput) [reply syft/workers/message_handler.py]: oops, done

**Comment by Nilanshrajput** [comment]
@gmuraru I added the issue #4634 .

**Review by gmuraru — APPROVED** [review]

**Comment by gmuraru** [comment]
LGTM!

**Review by replomancer — APPROVED** [review]
LGTM.

**Pull request merged**